DOC: add sections about big new features (CoW, string dtype) to 3.0.0 whatsnew notes (#61724)
pandas-dev/pandas: doc/source/whatsnew/v3.0.0.rst

Enhancements
~~~~~~~~~~~~

.. _whatsnew_300.enhancements.string_dtype:

Dedicated string data type by default
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Historically, pandas represented string columns with NumPy ``object`` data type.
This representation has numerous problems: it is not specific to strings (any
Python object can be stored in an ``object``-dtype array, not just strings) and
it is often not very efficient (both performance wise and for memory usage).

Starting with pandas 3.0, a dedicated string data type is enabled by default
(backed by PyArrow under the hood, if installed, otherwise falling back to

> NumPy). This means that pandas will start inferring columns containing string

Reviewer: ```suggestion
``object``-dtype backed by NumPy). This means that pandas will start inferring columns containing string
```

Author: With that edit it seems to suggest that it falls back to "object" dtype (as we currently use)? But it still has a StringDtype, just using numpy instead of pyarrow under the hood. So it is "backed by NumPy object dtype under the hood", so something like:

```suggestion
being backed by NumPy ``object``-dtype). This means that pandas will start inferring columns containing string
```

data as the new ``str`` data type when creating pandas objects, such as in
constructors or IO functions.

Old behavior:

.. code-block:: python

    >>> ser = pd.Series(["a", "b"])
    0    a
    1    b
    dtype: object

New behavior:

.. code-block:: python

    >>> ser = pd.Series(["a", "b"])
    0    a
    1    b
    dtype: str

The string data type that is used in these scenarios will mostly behave as NumPy
object would, including missing value semantics and general operations on these
columns.

The main characteristic of the new string data type:

- Inferred by default for string data (instead of object dtype)
- The ``str`` dtype can only hold strings (or missing values), in contrast to
  ``object`` dtype. (setitem with non string fails)
- The missing value sentinel is always ``NaN`` (``np.nan``) and follows the same
  missing value semantics as the other default dtypes.

Those intentional changes can have breaking consequences, for example when checking
for the ``.dtype`` being object dtype or checking the exact missing value sentinel.

> TODO add link to migration guide for more details

Author: Added the link (this page exists now)

.. seealso::

    `PDEP-14: Dedicated string data type for pandas 3.0 <https://pandas.pydata.org/pdeps/0014-string-dtype.html>`__

.. _whatsnew_300.enhancements.copy_on_write:

Copy-on-Write
^^^^^^^^^^^^^

The new "copy-on-write" behaviour in pandas 3.0 brings changes in behavior in
how pandas operates with respect to copies and views. A summary of the changes:

1. The result of *any* indexing operation (subsetting a DataFrame or Series in any way,

>    i.e. including accessing a DataFrame column as a Series) or any method returning a

Reviewer: ```suggestion
   e.g. accessing a DataFrame column as a Series) or any method returning a
```
